### PR TITLE
Apply weighted voting to reactions

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #9133
Closes: #9141
Closes: #9139 
Closes: #9261

## Description of Changes
- Applies topic-based weighted voting to thread reactions
- Updates the createTopic command to set weighted voting options
- Updates CreateContestManagerMetadata command to prevent creation of recurring contest if stake not enabled
- Adds migration that makes all non-general topics in a staked community to be staked

## Test Plan
- Go to a non-staked community
  - Create topic + thread + reaction – ensure no errors
- Enable stake in the community + buy stake + ensure you have > 0 stake
  - Create a thread in General, vote on it– ensure that the reaction weight is 1
  - Create a thread in a non-General topic, vote on it– ensure that the reaction weight is based on stake
  - Create a new topic– ensure no errors
  - Update a topic– ensure no errors

## Deployment Plan
N/A

## Other Considerations
- All new topics created in a staked community will be staked by default since the UI is not wired up. Once the UI is wired up, then the stake params from the UI will be used (stake/ERC20/none)– that code path is activated via the `FLAG_FARCASTER_CONTEST` env var

### CODE REVIEW

File: libs/model/src/services/stakeHelper.ts [hunk 1]
```diff
@@ -1,46 +1,73 @@
...
-    node.eth_chain_id,
-    [address],
-  );
-  const stakeBalance = stakeBalances[address];
-  if (BigNumber.from(stakeBalance).lte(0))
-    throw new InvalidState('Must have stake to upvote');
-
-  return commonProtocol.calculateVoteWeight(stakeBalance, stake.vote_weight);
+  const topic = await models.Topic.findByPk(topic_id);
+  mustExist('Topic', topic);
```
Comment (Rotorsoft) [inline libs/model/src/services/stakeHelper.ts:25]: can we load all these entities at once?

Comment (rbennettcw) [reply libs/model/src/services/stakeHelper.ts:48]: I've combined them into a single query.

File: libs/model/src/models/topic.ts [hunk 2]
```diff
@@ -29,6 +30,7 @@ export type TopicAttributes = {
   // associations
   community?: CommunityAttributes;
   threads?: ThreadAttributes[] | TopicAttributes['id'][];
+  chain_node?: ChainNodeAttributes;
```
Comment (Rotorsoft) [inline libs/model/src/models/topic.ts:33]: should this be called chain_node or something more specific around erc20 weights?

**Review by Rotorsoft — APPROVED** [review]
LGTM, just a small naming comment and wondering if we should refactor CreateTopic/UpdateTopic commands now that we are touching these files? Also, make sure thread lifecycle tests include topic/weight voting stuff with good coverage

**Review by masvelio — APPROVED** [review]
works according to test plan, however have one Q

File: libs/model/src/models/topic.ts (hunk 2 above)
Comment (rbennettcw) [reply libs/model/src/models/topic.ts:13]: I think it's fine. Community chain node is for stake, topic chain node is for erc20.

**Pull request merged**